rails-new.md

```diff
@@ -1,77 +1,126 @@
+
+cd ..
+gem install rails
+rails new bnb-library --database=postgresql
+mv bnb-library/* app
+exit
+```
+
+You will see a good deal of output telling you what Rails is creating for your new project. We are going to highlight some of the significant files, directories, and commands that you can find in the output of running `rails new`:
+
+- *gemfile* : This file lists the gem dependencies for your application. A gem is a Ruby software package, and a Gemfile allows you to manage your project's software needs. 
+- *app*: The app directory is where your main application code lives. This includes the models, controllers, views, assets, helpers, and mailers that make up the application itself. Rails gives you some application-level boilerplate for the MCV model to start out in files like `app/models/application_record.rb`, `app/controllers/application_controller.rb`, and `app/views/layouts/application.html.erb`.
+- *config*: This directory contains your application's configuration settings.
+- *config/routes*: Your application's route declarations live in this file.
+- *config/application.rb*: General settings for your application components are located in this file. 
+- *config/environments*: This directory is where configuration settings for your environments live. Rails includes three environments by default: `development`, `test`, and `production`. 
+- *config/database*: Database configuration settings live in this file, which is broken into four sections: `default`, `development`, `production`, and `test`. Thanks to the Gemfile that came with the `rails new bnb-library --database=postgresql`, which included the `pg` gem, our `config/database.yml` file has its adapter parameter set to postgresql already, specifying that we will use an postgresql database with this application. 
+- *db*: This folder includes a directory for database migrations called migrate, along with the `schema.rb` and `seeds.rb` files. `schema.db` contains information about your database, while `seeds.rb` is where you can place seed data for the database.
+
+## Step 3 - Starting Your Application
+
+We are going to want to interact with our app in the browser, but first, we must setup our database.
+
+We are going to add a *url* to our `database.yml` file which is going to tell our rails application a few things. Here is the breakdown...
+```
+postgres://DATABASE_USERNAME:DATABASE_PASSWORD@DATABASE_HOST:PORT/DATABASE_NAME
+```
+Under `development` we will add
+```
+development:
+  <<: *default
+  database: bnb_library_development
+  url: postgres://postgres:39dkdk3f93kkd93k20dl201kd83@postgres:5432
+```
+And `test`
+```
+test:
+  <<: *default
+  database: bnb_library_test
+  url: <%= ENV.fetch("DATABASE_URL") { "postgres://postgres:39dkdk3f93kkd93k20dl201kd83@postgres:5432" } %>
+```
+
+Now to setup the database we will need to start your server, as your postgresql service is tied in directly to the app
+
+```
+dip up
+```
+Then run the setup command in a seperate terminal
```

Comment:
```suggestion
Then run the setup command in a seperate terminal. This will spin up an instance of the `app` container to run the command in (as well as an instance of the `postgres` service that the `app` depends on)
```